Assets/Editor/TestResponeMergeUtil.cs

```diff
@@ -0,0 +1,117 @@
+namespace Shopify.Tests
+{
+    using System;
+    using NUnit.Framework;
+    using Shopify.Unity.SDK;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    [TestFixture]
+    public class TestResponseMergeUtil {
+        [Test]
+        public void TestDefaultMerge() {
+            Dictionary<string, object> responseA = GetResponseA();
+            Dictionary<string, object> responseB = GetResponseB();
+
+            ResponseMergeUtil merger = new ResponseMergeUtil();
+
+            Dictionary<string, object> merged = merger.Merge(responseA, responseB);
+
+            Assert.AreNotEqual(responseA, merged, "the merged dictionary is a dictionary. Is not responseA");
+            Assert.AreNotEqual(responseB, merged, "the merged dictionary is a dictionary. Is not responseB");
+            Assert.AreEqual(3, merged["fieldOnlyInA"], "Brought in field from A");
+            Assert.AreEqual("i am b", merged["fieldInBoth"], "Overwrote field in A from B");
+            Assert.AreEqual(1, ((List<int>) merged["fieldHasList"])[0], "Overwrote field in A from B that is a list");
+            Assert.AreEqual("monkey", ((Dictionary<string, string>) merged["fieldHasDictionary"])["keyA"], "Overwrote field in A from B that is a list");
+            Assert.AreEqual(responseB["fieldHasDictionary"], merged["fieldHasDictionary"]);
```

Comment:
Yes because what if it down the line someone implements copying/cloning `fieldHasDictionary` instead of the expected default outcome of set by reference. Although this should use `Object.ReferenceEquals` instead.